src/backend/mod.rs

```diff
@@ -595,14 +604,14 @@ extern "C" fn audiounit_output_callback(
     let (input_buffer, mut input_frames) = if !stm.core_stream_data.input_unit.is_null() {
         let input_buffer_manager = stm.core_stream_data.input_buffer_manager.as_mut().unwrap();
         assert_ne!(stm.core_stream_data.input_desc.mChannelsPerFrame, 0);
-        let input_channels = stm.core_stream_data.input_desc.mChannelsPerFrame as usize;
+        let input_channels = stm.core_stream_data.input_stream_params.channels() as usize;
```

Comment:
```suggestion
        let input_channels = input_buffer_manager.channel_count() as usize;
```

We should add a function in `BufferManager` to get this instead:
```rs
impl BufferManager {
    ...
    pub fn channel_count(&self) -> usize {
        self.output_channel_count
    }
    ...
```